cmd/proxy-to-grafana/proxy-to-grafana.go

```diff
@@ -0,0 +1,121 @@
+// proxy-to-grafana uses a reverse proxy which identifies users based on their originating tailscale node and creates / logs in users to grafana. Uses the auth proxy feature for grafana: https://grafana.com/docs/grafana/latest/auth/auth-proxy/
+//
+// Provide the env var TS_AUTHKEY to have this server automatically join your Tailnet.
+//
+// Use this grafana configuration to enable the auth proxy:
+
+// ```
+// [auth.proxy]
+// enabled = true
+// header_name = X-WEBAUTH-USER
+// header_property = username
+// auto_sign_up = true
+// whitelist = 127.0.0.1
+// headers = Name:X-WEBAUTH-NAME
+// enable_login_token = true
+// ```
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+
+	"inet.af/netaddr"
+	"tailscale.com/client/tailscale"
+	"tailscale.com/net/tsaddr"
+	"tailscale.com/tailcfg"
+	"tailscale.com/tsnet"
+)
+
+var (
+	hostname     = flag.String("hostname", "", "Tailscale hostname to serve on")
+	upstreamAddr = flag.String("upstream-addr", "", "Address of the upstream Grafana server, in host:port format")
+	tailscaleDir = flag.String("state-dir", "", "Directory in which to store the Tailscale auth state")
```

Comment:
Optional?

Reply:
You'd want it for certs

Reply:
But it should be automatic. I'd like this to be described like "alternate directory to use for Tailscale state storage. If empty, a default is used."

Reply:
👍